### ISSUE TITLE
Surrogate model framework: fit XGBoost to simulation surfaces for instant Greeks and reserves

### ISSUE THREAD

**Issue by joefarchione** [cavere-lang/actuarius#9]
## Summary

Use Cavere's batch simulation + AD to generate training surfaces over policy/economic state spaces, fit XGBoost surrogate models to the surface, and deploy the fitted trees back into Cavere via `TreeInfer` (see #8) for sub-millisecond inference. The entire pipeline stays in F# -- training via an XGBoost library exposed as an Analysis-layer function, inference via compiled GPU surfaces.

This extends Cavere's architecture in a natural direction: the simulation engine generates exact training data, the analysis layer fits the surrogate, and the surface system deploys it. No Python, no external toolchain.

## Architecture

```
Cavere batch simulation (GPU)     ->  SimResult arrays
Analysis layer (CPU, F#)          ->  XGBoost fit (grid features -> target)
Export fitted model                ->  flat float32 surface
Deploy via TreeInfer (GPU)         ->  instant per-policy inference
```

### Training

Run a synthetic batch where each "policy" is a grid node spanning the state space:

```fsharp
// Synthetic grid: 5K nodes covering the policy/economic state space
let premiums = [| 25_000f .. 25_000f .. 500_000f |]       // 20 levels
let ages     = [| 35f .. 5f .. 80f |]                      // 10 levels
let rates    = [| 0.01f .. 0.005f .. 0.06f |]              // 11 levels
let vols     = [| 0.10f .. 0.025f .. 0.35f |]              // 11 levels
// Cross product -> ~24K grid nodes, one GPU dispatch

let gridModel =
    AAAGenerator.buildBlockWithAAA
        AAACommon.defaultParams
        ProductRILA.flexGuardIncome4Index
        premiums ages rates vols  // batchInput arrays
```

One GPU dispatch produces exact Greeks (via AD) and option values at every grid node. The batch model already shares SLV rate paths and equity index paths across the block -- a synthetic grid works identically to a real policy block.

### Fitting

XGBoost training as an Analysis-layer function in F#:

```fsharp
let surrogateAnalysis = analysis {
    let! delta = observe "delta_spx"
    let! gamma = observe "gamma_spx"
    let! ov    = observe "option_value"
    let! av    = observe "account_value"

    // Fit surrogate: grid features -> target
    let deltaModel = Surrogate.fitTree features delta { MaxDepth = 6; NumTrees = 200; LearningRate = 0.1f }
    let gammaModel = Surrogate.fitTree features gamma { MaxDepth = 6; NumTrees = 200; LearningRate = 0.1f }
    let ovModel    = Surrogate.fitTree features ov    { MaxDepth = 8; NumTrees = 500; LearningRate = 0.05f }

    // Validation: holdout R-squared, max absolute error, tail accuracy
    let deltaR2 = Surrogate.validateHoldout deltaModel features delta 0.2f
    let tailAccuracy = Surrogate.validateTail ovModel features ov 0.05f  // bottom 5%

    // Export as Cavere surfaces
    Surrogate.exportSurface deltaModel "surfaces/delta_surrogate.bin"
    Surrogate.exportSurface gammaModel "surfaces/gamma_surrogate.bin"
    Surrogate.exportSurface ovModel    "surfaces/ov_surrogate.bin"

    return {| DeltaR2 = deltaR2; TailAccuracy = tailAccuracy |}
}
```

### Deployment

Fitted trees deploy as surfaces, queried via `tree_infer` inside production models:

```cavere
let delta_model = surface1d("surfaces/delta_surrogate.bin")
let ov_model = surface1d("surfaces/ov_surrogate.bin")

// Instant per-policy Greeks -- no simulation needed
let delta = tree_infer(delta_model, age, duration, premium, rate, vol, itm_ratio)
let ov = tree_infer(ov_model, age, duration, premium, rate, vol, itm_ratio)
```

### 1. Intraday Hedge Greeks

Full MC Greeks for a 50K-policy book take minutes. A fitted surrogate returns Greeks for the entire book in milliseconds. Re-fit weekly or when market moves exceed a calibrated threshold.

```
Morning: tree_infer() -> hedge ratios for all policies -> execute trades
Intraday: market moves -> tree_infer() with updated rates/vols -> adjust hedges
Weekly: re-run full MC grid -> re-fit surrogate -> validate against holdout
```

### 2. Real-Time Reserve Approximation

VM-21 CTE70 requires running thousands of scenarios. A surrogate trained on the full scenario set can approximate per-policy reserve contributions instantly. Use for:
- Daily reserve monitoring between full valuations
- What-if analysis: "what happens to reserves if rates drop 50bps?"
- Attribution: decompose reserve changes into rate, equity, lapse, and mortality components via surrogate feature importances

### 3. Dynamic Policyholder Behavior

Replace static decrement tables with state-dependent predictions. Train on historical experience studies:

```fsharp
// Training data: actual policyholder behavior from experience study
// Features: policy state at decision point
// Target: observed lapse/utilization/annuitization rate
let lapseModel = Surrogate.fitTree experienceFeatures observedLapseRates treeParams

// Deploy: lapse probability depends on current policy state
// tree_infer() inside the simulation loop -- evaluated every timestep
```

This closes the loop between experience studies (Phase 8 Output) and model assumptions (Phase 4 Input) -- the surrogate IS the updated assumption set.

### 4. Nested Stochastic Acceleration

Inner-loop valuations in nested stochastic (CTE within CTE, real-world outer / risk-neutral inner) are the computational bottleneck. Replace the inner loop with a surrogate:

```
Outer loop (real-world scenarios, GPU):
    At each future valuation date:
        -> tree_infer(surrogate, policy_state, economic_state) ~ inner-loop reserve
        Instead of: run 1000 inner-loop risk-neutral scenarios
```

Speedup is O(inner_scenarios) -- typically 1000x. The surrogate must be accurate in the tails (CTE is tail-sensitive), so validation focuses on the bottom 5-10% of the distribution.

### 5. Sensitivity Surface Exploration

AD gives exact point sensitivities. A surrogate gives the full sensitivity surface -- how Greeks vary across the entire state space. Use for:
- Identifying pockets of high gamma (hedging risk concentration)
- Stress testing: sweep economic states and read off reserve impacts
- Product design: evaluate how product feature changes (buffer level, cap rate) affect the Greek surface

### 6. Model Risk Quantification

Fit surrogates under different modeling assumptions (mortality tables, correlation structures, SLV calibrations). Compare surrogate surfaces to quantify model risk:

```fsharp
let baseModel = Surrogate.fitTree features resultsBase treeParams
let stressedMort = Surrogate.fitTree features resultsStressedMortality treeParams
let stressedCorr = Surrogate.fitTree features resultsStressedCorrelation treeParams

// Model risk = max divergence across the state space
let mortModelRisk = Surrogate.surfaceDivergence baseModel stressedMort features
let corrModelRisk = Surrogate.surfaceDivergence baseModel stressedCorr features
```

### F# XGBoost Integration

Expose XGBoost training as a `Surrogate` module in `Cavere.Actuarius`:

```fsharp
module Surrogate =
    type TreeParams = { MaxDepth: int; NumTrees: int; LearningRate: float32; SubsampleRate: float32 }
    type FittedTree  // opaque handle to trained model

    val fitTree: features: float32[,] -> targets: float32[] -> params: TreeParams -> FittedTree
    val predict: model: FittedTree -> features: float32[,] -> float32[]
    val validateHoldout: model: FittedTree -> features: float32[,] -> targets: float32[] -> holdoutFraction: float32 -> float32
    val validateTail: model: FittedTree -> features: float32[,] -> targets: float32[] -> tailFraction: float32 -> float32
    val exportSurface: model: FittedTree -> path: string -> unit
    val surfaceDivergence: model1: FittedTree -> model2: FittedTree -> features: float32[,] -> float32
```

Library options for the XGBoost binding:
- **ML.NET** -- Microsoft ML framework, has tree-based learners (FastTree, LightGBM), native .NET
- **XGBoost.Net** -- Direct XGBoost C API bindings for .NET
- **Custom implementation** -- gradient-boosted trees are not complex; a float32-native F# implementation avoids external dependencies and guarantees single-precision throughout

### Surface Export Format

Fitted trees serialize to the same flat float32 array format as other Cavere surfaces. A tree with N nodes:

```
[num_trees, tree_0_offset, tree_1_offset, ...,
 node_0_feature_idx, node_0_threshold, node_0_left, node_0_right, node_0_leaf_value,
 node_1_feature_idx, ...]
```

This packs into `surface1d` and transfers to GPU alongside rate curves and mortality tables. `TreeInfer` (#8) traverses this at kernel execution time.

### Validation Framework

The Analyst agent validates surrogates before deployment:

- **Accuracy**: R-squared > 0.999 on holdout set for Greeks, > 0.9999 for option values
- **Tail accuracy**: max absolute error in bottom 5% < 1% of CTE70
- **Monotonicity**: surrogate preserves known monotonic relationships (delta increases with spot, gamma peaks near ATM)
- **Extrapolation detection**: flag policies outside training grid convex hull
- **Staleness**: track assumption fingerprint (ModelManifest hash) -- re-fit when assumptions change

## Dependencies

- Depends on #8 (TreeInfer Expr case) for GPU-side deployment
- Core `Surrogate` module lives in `Cavere.Actuarius` (actuarial-specific grid designs, validation criteria)
- The `TreeInfer` Expr case and surface format are domain-agnostic (Cavere core)
- XGBoost fitting is an Analysis-layer operation -- CPU-bound, runs after simulation completes
